Source/Scenes/Onboarding/OnboardingInteractor.swift

```diff
@@ -0,0 +1,152 @@
+import Foundation
+
+// MARK: - Protocol
+protocol OnboardingInteractor: class {
+
+    // MARK: Typealiases
+
+    typealias Event = Onboarding.Event
+
+    // MARK: Requirements
+
+    func onViewDidLoad(request: Event.ViewDidLoad.Request)
+    func onSelectTopic(request: Event.DidSelectTopic.Request)
+    func onStartButtonDidPress(request: Event.StartButtonDidPress.Request)
+}
+
+extension Onboarding {
+
+    typealias Interactor = OnboardingInteractor
+
+    // MARK: - Declaration
+    class InteractorImp {
+
+        typealias Model = Onboarding.Model
+        typealias Event = Onboarding.Event
+
+        private let presenter: Presenter
+        private var sceneModel: Model.SceneModel
+        
+        init(presenter: Presenter) {
+            self.presenter = presenter
+            self.sceneModel = Model.SceneModel(numberForSelect: 3,
+                                               topics: [],
+                                               subtopics: [],
+                                               selectedTopicsIds: [],
+                                               isStartEnabled: false)
+        }
+    }
+}
+
+// MARK: - Private
+private extension Onboarding.InteractorImp {
+    func fetchTopics() {
+        DispatchQueue.global(qos: .background).asyncAfter(deadline: .now() + 0.2) {
```

Comment:
Move this and all related things (like fetching, updating, reloading and so on) to fetcher worker